### PR TITLE
Addreportfunction

### PR DESCRIPTION
**Pull request by awirb**
* Add generate markdown report
* Add function to plot all peak fields
* Add function to create one specific peak file
* Add docu for out3Plot, out3SimpPlot, log2Report

### CODE REVIEW

File: avaframe/in3Utils/fileHandlerUtils.py [hunk 1]
```diff
@@ -61,6 +61,48 @@ def readLogFile(logName, cfg=''):
...
+                mass.append(float(line.split()[3]))
+
+    # Save to dictionary
+    logDict = {'time': np.asarray(time), 'mass': np.asarray(mass),
+               'entrMass': np.asarray(entrMass)}
+
+    # Set values in dictionary
+    parameterDict['release mass [kg]'] = logDict['mass'][0]
+    parameterDict['final time step [s]'] = logDict['time'][-1]
+    parameterDict['current mass [kg]'] = logDict['mass'][-1]
```
Comment (fso42) [inline avaframe/in3Utils/fileHandlerUtils.py:101]: What is locDict needed for? Wouldn't it be possible just use
`
parameterDict['release mass [kg]'] = np.asarray(mass)[0]
`

File: avaframe/in3Utils/fileHandlerUtils.py [hunk 2]
```diff
@@ -213,19 +255,21 @@ def makeSimDict(inputDir, varPar=''):
...
 
     # Make dictionary of input data info
-    data = {'files': [], 'names': [], 'resType': [], 'simType': [],
+    data = {'files': [], 'names': [], 'resType': [], 'simType': [], 'simName' : [],
             'modelType' : [], 'releaseArea': [], 'cellSize': [], varPar: []}
 
     for m in range(len(datafiles)):
         data['files'].append(datafiles[m])
         name = os.path.splitext(os.path.basename(datafiles[m]))[0]
+        data['simName']
```
Comment (fso42) [inline avaframe/in3Utils/fileHandlerUtils.py:264]: What is this needed for?

File: avaframe/in3Utils/fileHandlerUtils.py [hunk 3]
```diff
@@ -213,19 +255,21 @@ def makeSimDict(inputDir, varPar=''):
...
         data['names'].append(name)
         nameParts = name.split('_')
         data['releaseArea'].append(nameParts[0])
         data['simType'].append(nameParts[1])
         data['modelType'].append(nameParts[2])
         data[varPar].append(nameParts[3])
         data['resType'].append(nameParts[4])
+        data['simName'].append(nameParts[0] + '_' + nameParts[1] + '_' + nameParts[2] + '_' + nameParts[3])
```
Comment (fso42) [inline avaframe/in3Utils/fileHandlerUtils.py:272]: Maybe use `'_'.join(nameParts[0:3]) ` ?

File: avaframe/out3SimpPlot/outPlotAllPeak.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        cmap, _, _, norm, ticks = makePalette.makeColorMap(
+            cmapPres, np.amin(data), np.amax(data), continuous=contCmap)
+        im1 = ax1.imshow(data, cmap=cmap, extent=[0, Lx, 0, Ly], origin='lower', aspect=nx/ny)
+        addColorBar(im1, ax1, ticks, unit)
+        title = str('%s' % name)
+        ax1.set_title(title)
+        ax1.set_xlabel('x [m]')
+        ax1.set_ylabel('y [m]')
+        plotName = os.path.join(outDir, '%s.%s' % (name, outputFormat))
+        fig.savefig(plotName)
```
Comment (fso42) [inline avaframe/out3SimpPlot/outPlotAllPeak.py:67]: Could you use the saveAndOrPlot function from plotUtils -> to honor global flags about saving/plotting...

**Review by fso42 — CHANGES_REQUESTED** [review]
Regarding report images: 
right now they are just file paths in the markdown, correct?
Should we copy them to a directory that lives right by the markdown file? So someone could just copy the report?

Please address some of the flake8/autopep issues...

File: avaframe/com1DFA/com1DFA.py [hunk 5]
```diff
@@ -168,14 +168,19 @@ def runSamos(cfg, avaDir):
     if flagEntRes:
         entrainmentArea = os.path.splitext(os.path.basename(ent))[0]
         resistanceArea = os.path.splitext(os.path.basename(res))[0]
-
+    else:
+        entrainmentArea = ''
+        resistanceArea = ''
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:173]: Yes, default is empty, done

File: avaframe/com1DFA/com1DFA.py [hunk 6]
```diff
@@ -240,6 +245,17 @@ def runSamos(cfg, avaDir):
             copyReplace(workFile, workFile, '##VARPAR##', cfgPar['varPar'])
             copyReplace(workFile, workFile, '##VALUE##', defValues[cfgPar['varPar']])
             execSamos(samosAT, workFile, avaDir, fullOut, logName)
+
+            # Create dictionary
+            dictNull = {}
+            dictNull = {'headerLine' : 'com1DFA simulation', 'simName' : logName, 'simParameters' : {'release area' : relName, 'entrainment area' : entrainmentArea, 'resistance area' : resistanceArea,
+                        'parameter variation on' :'', 'parameter value' : '', 'Mu' : defValues['Mu'], 'release thickness [m]' : defValues['RelTh']}, 'Release area' : {'release area scenario' : relName},
+                       'Entrainment area' : {'entrainment area scenario' : entrainmentArea}, 'Resistance area' : {'resistance area scenario' : resistanceArea}}
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:253]: Done

File: avaframe/com1DFA/com1DFA.py [hunk 7]
```diff
@@ -285,9 +317,25 @@ def runSamos(cfg, avaDir):
                 copyReplace(workFile, workFile, '##VALUE##', defValues['Mu'])
                 execSamos(samosAT, workFile, avaDir, fullOut, logName)
 
+                # Create dictionary
+                dictST = {}
+                dictST = {'headerLine' : 'com1DFA simulation', 'simName' : logName, 'simParameters' : {'release area' : relName, 'entrainment area' : entrainmentArea, 'resistance area' : resistanceArea,
+                            'parameter variation on' :'', 'parameter value' : '', 'Mu' : defValues['Mu'], 'release thickness [m]' : defValues['RelTh']}, 'Release area' : {'release area scenario' : relName},
+                           'Entrainment area' : {'entrainment area scenario' : entrainmentArea}, 'Resistance area' : {'resistance area scenario' : resistanceArea}}
+
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:325]: Done

File: avaframe/com1DFA/com1DFA.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFA.py:173]: Done

File: avaframe/com1DFA/com1DFACfg.ini [hunk 8]
```diff
@@ -38,3 +38,22 @@ varRunCint = vary
...
+# release snow density [kgm-3]
+rhoRelease = 200
+# End time [s] of simulation if stop criterion is not met
+finalTime = 400
+# time step used in computations
+timeStep = 0.1
+# entrainment snow density [kgm-3]
+rhoEntrainment = 100
+# entrainment snow thickness [m]
+entH = 0.3
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFACfg.ini:53]: ok I excluded them for now

File: avaframe/com1DFA/com1DFACfg.ini [hunk 9]
```diff
@@ -38,3 +38,22 @@ varRunCint = vary
 Mu = 0.155
 # standard value for release area thickness
 RelTh = 1.000
+
+
+[REP]
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com1DFA/com1DFACfg.ini:43]: Done

File: avaframe/in3Utils/fileHandlerUtils.py (hunk 1 above)
Comment (awirb) [reply avaframe/in3Utils/fileHandlerUtils.py:101]: Done

File: avaframe/in3Utils/fileHandlerUtils.py (hunk 3 above)
Comment (awirb) [reply avaframe/in3Utils/fileHandlerUtils.py:272]: Done

File: avaframe/in3Utils/fileHandlerUtils.py (hunk 2 above)
Comment (awirb) [reply avaframe/in3Utils/fileHandlerUtils.py:264]: simName is e.g. used in the generate report function and the plot all peak files function

Comment (fso42) [reply avaframe/in3Utils/fileHandlerUtils.py:260]: But this line does not do anything!? Or is it meant for initialising the variable?

Comment (awirb) [reply avaframe/in3Utils/fileHandlerUtils.py:260]: Correct, setting simName appears some lines later, so I removed it here

File: avaframe/out3SimpPlot/outPlotAllPeak.py (hunk 4 above)
Comment (awirb) [reply avaframe/out3SimpPlot/outPlotAllPeak.py:73]: the cfgPaths variable is not set globally so i would move this as a request for a future PR?

Comment (fso42) [reply avaframe/out3SimpPlot/outPlotAllPeak.py:73]: jup, move it to an issue

**Review by fso42 — APPROVED** [review]

**Pull request merged**